### PR TITLE
Add mixed dict and vec types

### PR DESCRIPTION
**Pull request by BackEndTea**

### CODE REVIEW

File: src/Psl/Type/Internal/MixedDictType.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+final class MixedDictType extends Type\Type
+{
+    /**
+     * @throws CoercionException
+     *
+     * @return array<array-key, mixed>
+     */
+    public function coerce(mixed $value): array
+    {
+        if (! is_iterable($value)) {
```
Comment (BackEndTea) [inline src/Psl/Type/Internal/MixedDictType.php:25]: I just noticed the PHP-cs fixed did not pick up on this not being imported in my dev, or in CI.

**Comment by azjezz** [comment]
LGTM, can you rebase this against 2.1.x? ( see #361 )

**Comment by jrmajor** [comment]
What would be the difference between `T\vec(T\mixed())` and `T\mixed_vec()`?

**Comment by BackEndTea** [comment]
> What would be the difference between `T\vec(T\mixed())` and `T\mixed_vec()`?

Better performance, this won't loop over the entire array if its not needed. 

Se also https://github.com/azjezz/psl/issues/359

**Review by azjezz — APPROVED** [review]

**Comment by azjezz** [comment]
Thanks @BackEndTea

**Comment by BackEndTea** [comment]
Thanks @azjezz !

**Pull request merged**